weave/trace_server/trace_server_converter.py

```diff
@@ -0,0 +1,116 @@
+import typing
+
+from pydantic import BaseModel
+
+from . import refs_internal as ri
+
+
+def universal_ext_to_int_ref_converter(
+    obj: typing.Any, convert_ext_to_int_project_id: typing.Callable[[str], str]
+) -> typing.Any:
+    """Takes any object and recursively replaces all external references with
+    internal references. The external references are expected to be in the
+    format of `weave:///entity/project/...` and the internal references are
+    expected to be in the format of `weave-internal:///project_id/...`.
```

Comment:
Is it actually `weave-trace-internal:///...`

Reply:
damn. actually used code-pilot to write this and still missed something.